app/controllers/api/cloud_templates_controller.rb

```diff
@@ -11,7 +11,10 @@ def import_resource(_type, _id, data = {})
 
       ems_dst   = resource_search(data['dst_provider_id'], :providers)
       ems_src   = resource_search(data['src_provider_id'], :providers)
-      src_image = resource_search(data['src_image_id'],    :providers, collection_class(:templates))
+      src_image = resource_search(data['src_image_id'], :providers, collection_class(:templates))
+      resource_search(data['obj_storage_id'], :providers) if data['obj_storage_id'].present?
+      resource_search(data['bucket_id'], :cloud_object_store_containers) if data['bucket_id'].present?
+      resource_search(data['disk_type_id'], :cloud_volume_types) if data['disk_type_id'].present?
```

Comment:
So this raises a 403 exception or does it return `nil` and expect you to raise a RecordNotFound or BadRequestError exception?  Should we raise a BadRequestError if this returns `nil` e.g. the id they passed doesn't exist like we do down below for the src_image?

Reply:
good question. had to research.

`resource_search` raises a `BadRequestError` if the id is `nil` (via `validate_id`) or if the search is not found (no privs).

So this is good. (and the `if data[]` check is necessary too)

Reply:
@kbrock I believe data can't be `nil` as it has a default value set to empty hash here: https://github.com/ManageIQ/manageiq-api/blob/3415b126bcecd64af8cbec1dcb5a70f999b87485/app/controllers/api/cloud_templates_controller.rb#L7